src/main/java/mil/dds/anet/config/AnetConfiguration.java

```diff
@@ -146,6 +171,69 @@ public void setDictionary(Map<String, Object> dictionary) {
     this.dictionary = Collections.unmodifiableMap(dictionary);
   }
 
+  @SuppressWarnings("unchecked")
+  public void loadDictionary() {
+    // Read and set anet-dictionary
+    final File file = new File(System.getProperty("user.dir") + "/anet-dictionary.yml");
+    try (final InputStream inputStream = new FileInputStream(file)) {
+      final Map<String, Object> dictionary = yamlMapper.readValue(inputStream, Map.class);
+      final Map<String, Object> settings = (Map<String, Object>) dictionary.get("dictionary");
+      this.setDictionary(settings);
+      // Check the dictionary
+      final JsonNode jsonNodeDictionary = checkDictionary();
+      logger.info("dictionary: {}", yamlMapper.writeValueAsString(jsonNodeDictionary));
+    } catch (IOException e) {
+      logger.error("ANET dictionary [anet-dictionary.yml] not found");
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public void reloadDictionary() {
+    final Map<String, Object> previousDictionary = this.getDictionary();
+    try {
+      final File file = new File(System.getProperty("user.dir") + "/anet-dictionary.yml");
+      try (final InputStream inputStream = new FileInputStream(file)) {
+        final Map<String, Object> dictionary = yamlMapper.readValue(inputStream, Map.class);
+        final Map<String, Object> settings = (Map<String, Object>) dictionary.get("dictionary");
+        this.setDictionary(settings);
+      } catch (IOException e) {
+        logger.error("ANET dictionary [anet-dictionary.yml] not found");
+      }
+      // Check the dictionary
+      checkDictionary();
+    } catch (IllegalArgumentException e) {
+      this.setDictionary(previousDictionary);
+      throw new WebApplicationException(("Invalid dictionary in the configuration"));
```

Comment:
probably premature to throw WebApplicationException as we are not sure who is calling this method yet. I would just call `loadDictionary` whenever reloading the dictionary is needed and handle the exceptions where the call is happening

Reply:
The same here , I've changed the way . I think it is ok now .